### DISCUSSION TITLE
Difference between supabase query and other ORM (prisma)

### DISCUSSION THREAD

**Discussion by primostasis123** [supabase/supabase#27289] [Questions]
Hi,

I'm currently facing an issue where using Prisma for my queries matches the table list in Supabase, but using Supabase queries (Supabase URL and Supabase anon key) does not match. There might be something wrong with my code. For context, I am able to insert data into Supabase using Supabase insert, but the problem is that the query results do not match.

im using Next JS 13.5.6

```
//supabase.ts

import { createClient } from '@supabase/supabase-js'

export default createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL!,
  process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!
)
```

```
"use server";
import supabase from '@/lib/supabase'

export const getBooks = async () => {
  const { data } = await supabase.from('books').select("*")
  return data
};

export const createBooks = async ({
  title,
  author,
  published_date,
  genre,
}: IBook) => {

  const date = new Date(published_date);
  const isoDate = date.toISOString(); 
  
  const { data, error } = await supabase.from('books').insert([
    { title, author, published_date: date, genre }
  ])
  
};
};
```

**Comment by GaryAustin1** [comment]
Might help to show how they don't match.
If you have RLS enabled then you will also need a policy for supabase-js, but that does not apply to the direct database port from Prisma (in the default case).

**Reply by primostasis123** [reply]
> Might help to show how they don't match. If you have RLS enabled then you will also need a policy for supabase-js, but that does not apply to the direct database port from Prisma (in the default case).

RLS is disabled

**Reply by GaryAustin1** [reply]
OK... how don't they match?
If you mean order, there is no default order and it could be different from one request to another depending on what part of disk is cached.

**Reply by GaryAustin1** [reply]
Also next.js caches get URL's in many cases, so you could have a cached select result if you did inserts after the initial select.

**Reply by primostasis123** [reply]
> OK... how don't they match? If you mean order, there is no default order and it could be different from one request to another depending on what part of disk is cached.

the numbers of data does not match..

I found a work around using this in server action . now the number of data is match

```
import { createServerActionClient } from "@supabase/auth-helpers-nextjs"
import { cookies } from "next/headers"

export const supabase = createServerActionClient({
  cookies,
})

```

The problem is that NEXT_PUBLIC_SUPABASE_URL and NEXT_PUBLIC_SUPABASE_ANON_KEY are set automatically by the library from the .env file. If I remove NEXT_PUBLIC, it will not work. However, I believe NEXT_PUBLIC is not safe, right? do you know some work around?

**Reply by GaryAustin1** [reply]
Anon key and URL are safe with public.  Only service role should not be used in public.

**Reply by GaryAustin1** [reply]
Your RLS would be used to protect data to signed in users versus anon if needed.  If you don't have signed in users, or don't want anyone to read the tables then you would not have RLS policies with RLS enabled.  Then use service_role client SERVER SIDE only and not using an SSR client if that is what you are doing.  You would use a supabase-js client with service_role.
